config/env_spec_request_test.go

```diff
@@ -365,12 +365,12 @@ func TestIsAuthenticated(t *testing.T) {
 			envoyReq := testutil.NewEnvoyRequest(http.MethodGet, test.path, nil, nil)
 			req := NewEnvironmentSpecRequest(&testAuthMan{}, specExt, envoyReq)
 
-			if req.IsAuthenticated() {
-				t.Fatalf("IsAuthenticated should be false")
+			if req.IsAuthenticated() == nil {
```

Comment:
+1

Or it should return both a bool and error?

Reply:
Actually, now I'm not sure why we changed this function. We only call it from one place (I believe)... could not we have left it as a boolean and just generate the error at the caller if the result is false?

Reply:
Renamed it to Authenticate() and replaced the usages as suggested by Eric. 
 I think Authenticate() returning an error is more axiomatic way of doing things instead of caller doing the work simply because there is only one caller `so far`.
Thanks!